Python/bltinmodule.c

```diff
@@ -2551,14 +2574,41 @@ zip_next(zipobject *lz)
 }
 
 static PyObject *
+zip_len(zipobject *lz)
+{
+    Py_ssize_t tuplesize = lz->tuplesize;
+    Py_ssize_t length_hint = PY_SSIZE_T_MAX;
+    Py_ssize_t i;
+
+    if (tuplesize == 0) {
+        Py_INCREF(_PyLong_Zero);
+        return _PyLong_Zero;
+    }
+
+    for (i = 0; length_hint > 0 && i < PyTuple_GET_SIZE(lz->ittuple); i++) {
+        PyObject *it = PyTuple_GET_ITEM(lz->ittuple, i);
+        Py_ssize_t it_len = PyObject_LengthHint(it, 0);
+
+        if (it_len < length_hint)
+            length_hint = it_len;
+    }
+
+    if (length_hint == -1)
+        return NULL;
+
```

Comment:
Actually `it_len == PY_SSIZE_T_MAX` should be valid because if there's no iterator then the function short-circuits immediatly.

However the issue was recently challenged and eventually closed as "no fix" so I haven't continued here.